lib/pbench/server/api/resources/query_apis/datasets/table_of_content.py

```diff
@@ -53,7 +53,8 @@ def assemble(self, json_data: JSON, context: CONTEXT) -> JSON:
             "parent": '/1-default'
         }
         """
-        self.parent = json_data.get("parent")
+        # Copy parent directory metadata to CONTEXT for postprocessor
+        context["parent"] = json_data.get("parent")
```

Comment:
This is good; so now we shouldn't need the `TableOfContent.parent` class attribute, so you should remove it...and all the lingering references to it need to be replaced.  (I would set up a local variable for this function, like it does for `run_id`, and use that.)